apiserver/facades/client/modelconfig/modelconfig_test.go

```diff
@@ -150,89 +160,50 @@ func (s *modelconfigSuite) TestBlockChangesModelSet(c *gc.C) {
 	s.assertModelSetBlocked(c, args, "TestBlockChangesModelSet")
 }
 
-func (s *modelconfigSuite) TestModelSetCannotChangeAgentVersion(c *gc.C) {
-	api, _ := s.getAPI(c)
-	old, err := config.New(config.UseDefaults, coretesting.FakeConfig().Merge(coretesting.Attrs{
-		"agent-version": "1.2.3.4",
-	}))
-	c.Assert(err, jc.ErrorIsNil)
-	s.backend.old = old
-	args := params.ModelSet{
-		Config: map[string]interface{}{"agent-version": "9.9.9"},
-	}
-	err = api.ModelSet(context.Background(), args)
-	c.Assert(err, gc.ErrorMatches, "agent-version cannot be changed")
-
-	// It's okay to pass config back with the same agent-version.
-	result, err := api.ModelGet(context.Background())
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(result.Config["agent-version"], gc.NotNil)
-	args.Config["agent-version"] = result.Config["agent-version"].Value
-	err = api.ModelSet(context.Background(), args)
-	c.Assert(err, jc.ErrorIsNil)
-}
-
-func (s *modelconfigSuite) TestModelSetCannotChangeCharmHubURL(c *gc.C) {
```

Comment:
These should be moved to the service as a smoke test to ensure the validators have been wired in. The individual validators can be unit tested fully, but we should have a smoke test for each one at the service level